spring-ai-model/src/main/java/org/springframework/ai/chat/memory/ChatMemoryRepository.java

```diff
@@ -16,30 +16,24 @@
 
 package org.springframework.ai.chat.memory;
 
-import java.util.List;
-
 import org.springframework.ai.chat.messages.Message;
 
+import java.util.List;
+
 /**
- * The ChatMemory interface represents a storage for chat conversation history. It
- * provides methods to add messages to a conversation, retrieve messages from a
- * conversation, and clear the conversation history.
+ * A repository for storing and retrieving chat messages.
  *
- * @author Christian Tzolov
+ * @author Thomas Vitale
  * @since 1.0.0
  */
-public interface ChatMemory {
-
-	// TODO: consider a non-blocking interface for streaming usages
+public interface ChatMemoryRepository {
 
-	default void add(String conversationId, Message message) {
-		this.add(conversationId, List.of(message));
-	}
+	List<String> findConversationIds();
 
-	void add(String conversationId, List<Message> messages);
+	List<Message> findMessages(String conversationId);
```

Comment:
findByConversationId instead?  just thinking of spring data repo conventions for finders.

Reply:
Done